README.md
# infinitus-custom-contacts
infinitus-custom-contacts

[中文文档](https://github.com/vkingw/custom-contacts/blob/master/README_zh.md)

威发微服务选人组件/Wafer microservice selection component

[![GitHub license](https://img.shields.io/badge/license-MIT-blue.svg)](https://github.com/vkingw/custom-contacts)
[![npm](https://img.shields.io/npm/v/react-contacts.svg)](https://www.npmjs.com/package/infinitus-custom-contacts)
[![NPM downloads](https://img.shields.io/npm/dm/react-contacts.svg)](https://www.npmjs.com/package/infinitus-custom-contacts)

## Example

![Example](./example.png)

## How to use

### install

`yarn add infinitus-custom-contacts`

### React 

```js
import Contacts from 'infinitus-custom-contacts';

<Contacts {...props}/>

```

Properties  | Description | Type | Default Values
------------- | ------------- | --------------| ------------- 
loading | Loading status | bool | false
searchResult | Query user data return results （[User & Search Result Data](###User & Search Result Data)） | object | { records: []}
handleSearchUser | Handle search user function | func | function(page, jobs, name, rank, rankClassification), page:Number  jobs:obj  name:String rank:obj rankClassification:obj
updateSelectUsers | Update user list when select user | func | function(userSelected)
defaultUserSelected   | default user selected | array | [{id:'abc',name:'CCC'}] 
numberColor   | total number color | sring | #E65653
totalShowText   | totalShowText | string | 共选择了$个
radioShowText   | when user selected on radio, show tip text. | string | '已经选择' 
jobsText   | jobs text | string | '岗位' 
nameText   | name text | string | '姓名' 
rankText   | rank text | string | '职级' 
workNumberNumber   | workNumberNumber text | string | '工号' 
functionText   | function text | string | '岗位' 
namePlaceholder   | name placeholder | string | '请输入搜索姓名' 
jobsPlaceholder   | jobs placeholder | string | '请选择岗位' 
rankPlaceholder   | rank placeholder | string | '请选择职级' 
rankTipPlaceholder   | rankTip placeholder | string | '请选择职级分类或级别' 
jobsData   | jobs select data | string | [] 
jobsValueKey   | jobs select key  | string | 'id' 
jobsNameKey   | jobs select name | string | 'name' 
rankData   | rank select  data | string | [] 
rankValueKey   | rank select key  | string | 'id' 
rankNameKey   | rank select name | string | 'name' 
rankClassificationData   | rankClassification select data | string | [] 
rankClassificationValueKey   | rankClassification select key | string | 'id' 
rankClassificationNameKey   | rankClassification select name | string | 'name'  
tableColumnsKey   | The query result table columns corresponds to dataIndex, please note that the first name and data echo also correspond | array |  ['name', 'workNumberNumber', 'fun', 'rank', 'jobs'],
tableRowKey   | table row key，This value corresponds to the data echo| string | 'id' 
emptyTip   | table empty tip | string | '什么都没有哦~'
tableCheckboxDisabled   | The set of keys that need to be forbidden to be selected in the table | array | []

### 职级分类或级别数据结构

````
[
      {
        id: 1,
        parentId: 0,
        children: [
          {
            id: 3,
            parentId: 1,
            children: [
              {
                id: 4,
                parentId: 3,
                children: [
                  {
                    id: 5,
                    parentId: 4,
                    children: [],
                    name: '院校农信',
                  },
                ],
                name: '高新农信',
              },
            ],
            name: '潍坊农信',
          },
        ],
        name: '山东农信',
      },
  ]
````

### User & Search Result Data

  const columns = [
    {
      title: nameText,
      dataIndex: 'name',
    },
    {
      title: workNumberNumber,
      dataIndex: 'workNumberNumber',
    },
    {
      title: functionText,
      dataIndex: 'fun',
    },
    {
      title: rankText,
      dataIndex: 'rank',
    }, {
      title: jobsText,
      dataIndex: 'jobs',
    }
  ];

```

 {
      records: [
        {
          id: 1,
          name:'abc',
          workNumberNumber:12,
          fun:'aaa',
          rank:'bbb',
          jobs:'ttt' 
        },
	   ......
      ],
      total: 11,
      size: 10,
      current: 1,
      searchCount: true,
      pages: 1,
    },

```

### Development

````
$ git clone https://github.com/wafersystems/react-contacts.git
$ yarn
$ yarn start

````


CHANGELOG.md
# Changelog
All notable changes to this project will be documented in this file.

The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).


## [0.0.16] - 2020-12-23

### Added

- none

### Changed

- modify user list add rolling

### Removed

- none

&nbsp;

## [0.0.15] - 2020-12-10

### Added

- none

### Changed

- modify a table's header style change.

### Removed

- none

&nbsp;

## [0.0.14] - 2020-11-20

### Added

- none

### Changed

- add tableCheckboxDisabled. The set of keys that need to be forbidden to be selected in the table

### Removed

- none

&nbsp;

## [0.0.12] - 2020-11-04

### Added

- none

### Changed

- fix rankClassification check error.

### Removed

- none

&nbsp;



src/example/index.js
import React from 'react';
import ReactDOM from 'react-dom';
import Contacts from '../components/contacts';
import {  userList,us,defaultSelect,tree } from '../mockData';

const updateSelectUsers = userList => {
  console.log(userList)
};

const updateSelectDept = deptList => {
  console.log(deptList)
};

const jobsData = [
  {id:1,name:'abc'},
  {id:2,name:'abc1'},
  {id:3,name:'abc1'},
  {id:41,name:'abc1'},
]

ReactDOM.render(<div className="App"
                     style={{ 'padding': '10px', 'margin': '10px', height: '100%' }}>
  <header className="App-header">
    <h1 className="App-title">Welcome to React</h1>
  </header>
  <p className="App-intro">
    To get started, edit <code>src/App.js</code> and save to reload.
  </p>
  <div style={{ height: '100%', width:690 }}>
    <Contacts loading={false}
              defaultUserSelected={defaultSelect}
              searchResult={us}
              updateSelectUsers={updateSelectUsers} handleSearchUser={() => userList} updateSelectDept={updateSelectDept}
              rankData={jobsData} jobsData={jobsData} rankClassificationData={tree}
              tableColumnsKey={['chsName', 'accountName', 'orgName', 'gradeType', 'positionName']}
              tableRowKey={'accountName'} tableCheckboxDisabled={['5601']}
    />
  </div>
</div>, document.getElementById('root'));


src/components/right.js
import React, { useEffect, useState } from 'react';
import { Card, Checkbox, Radio, Col, Input, List, message, Pagination } from 'antd';
import styles from './contacts.less';

const { Search } = Input;

export default ({
                  selectAllText, searchUserPlaceholder, deptSearch, userData,
                  handleSearch, handleSearchUser, deptId, updateSelectUsers, debug = false,
                  setOnSearch, nameKey, setNameKey, selectUser, setSelectUser,
                  userNameKey, radio
                }) => {

  const [selectAll, setSelectAll] = useState(false);

  // 当列表数据发生变化时，重新计算全选
  useEffect(() => {
    calculateSelectAll(selectUser);
  }, [userData]);

  // 当选中人数据发生变化，重新计算全选
  useEffect(() => {
    calculateSelectAll(selectUser);
  }, [selectUser]);

  /**
   * 姓名搜索为空时处理
   * @param e
   */
  const handleSearchChange = e => {
    if (!e.target.value) {
      setOnSearch(false);
      setNameKey(null);
    }
  };

  /**
   * 判断用户是否选择
   * @param data
   * @return {boolean}
   */
  const isUserCheck = data => {
    const result = selectUser.find(value => value.userId === data.userId);
    return !!result;
  };

  /**
   * 点击用户列表的回调
   * @param e
   */
  const onUserCheck = e => {
    const {
      target: { checked, data },
    } = e;
    const tmp = [];
    let newSelectUser;
    if (checked) {
      tmp.push(data);
      newSelectUser = selectUser.concat(tmp);
    } else {
      const result = selectUser.filter(value => value.userId !== data.userId);
      newSelectUser = result.concat(tmp);
    }
    setSelectUser(newSelectUser);
    updateSelectUsers(newSelectUser);
    if (checked) {
      calculateSelectAll(newSelectUser);
    } else {
      setSelectAll(false);
    }
  };

  /**
   * Radio时点击用户列表的回调
   * @param e
   */
  const onUserRadioCheck = e => {
    const {
      target: { checked, data },
    } = e;
    let newSelectUser = [];
    if (checked) {
      newSelectUser.push(data);
    } else {
    }
    setSelectUser(newSelectUser);
    updateSelectUsers(newSelectUser);
  };

  /**
   * 翻页查询处理
   * @param page
   */
  const onPageChange = page => {
    if (debug) {
      window.console.log(page)
    }
    if (handleSearchUser) {
      handleSearchUser(page, nameKey, deptId);
      setOnSearch(true);
    } else {
      message.error('search function not found.');
    }
  };

  /**
   * 计算是否全部选中
   * @param newSelectUser
   */
  const calculateSelectAll = (newSelectUser) => {
    let tmp = [];
    userData.records.forEach(value => {
      tmp.push(value);
    });
    if (tmp.length === 0) {
      setSelectAll(false);
      return;
    }
    let count = 0;
    tmp.forEach(val => {
      const result = newSelectUser.find(valUser => val.userId === valUser.userId);
      if (result) {
        count += 1;
      }
    });
    setSelectAll(count === tmp.length);
  };

  /**
   * 点击用户全选的回调
   * @param e
   */
  const onCheckAll = e => {
    const {
      target: { checked },
    } = e;
    setSelectAll(checked);
    const tmp = [];
    userData.records.forEach(value => {
      tmp.push(value);
    });
    let newSelectUser = [];
    if (checked) {
      // 如果是选中，遍历添加，重复的不添加
      tmp.forEach(val => {
        const result = selectUser.find(valUser => val.userId === valUser.userId);
        if (!result) {
          newSelectUser.push(val);
        }
      });
      newSelectUser = selectUser.concat(newSelectUser);
    } else {
      // 不选中的遍历删除
      selectUser.forEach(val => {
        const result = tmp.find(valUser => val.userId === valUser.userId);
        if (!result) {
          newSelectUser.push(val);
        }
      });
    }
    updateSelectUsers(newSelectUser);
    setSelectUser(newSelectUser);
  };

  return (
    <Col span={24} className={styles.treeLeft}>
      <Card className={styles.card}>
        <List
          size="small"
          bordered={false}
          dataSource={userData.records}
          split={false}
          renderItem={item => {
            return (
              <List.Item>
                <div className={styles.itemDiv}>
                  {radio &&
                  <Radio
                    className={styles.checkbox}
                    checked={isUserCheck(item)}
                    data={item}
                    onChange={onUserRadioCheck}
                  >
                    {item[userNameKey]}
                  </Radio>
                  }
                  {!radio &&
                  <Checkbox
                    className={styles.checkbox}
                    data={item}
                    checked={isUserCheck(item)}
                    onChange={onUserCheck}
                  >
                    {item[userNameKey]}
                  </Checkbox>}
                  <div className={styles.deptName}>{item.deptName}</div>
                </div>
              </List.Item>
            );
          }}
        />
      </Card>
      <div className={styles.pagination}>
        <Pagination
          className={styles.pageNoe}
          simple
          current={userData.current || 1}
          pageSize={userData.size}
          total={userData.total}
          onChange={onPageChange}
        />
      </div>
    </Col>)
}


src/components/search.js
import {Col, Form, Input, Row, Select, TreeSelect, Radio} from "antd";
import React, {useState} from "react";

import styles from './contacts.less';

const {Option} = Select;
const {TreeNode} = TreeSelect;

const makeTreeNode = (rankClassificationData, rankClassificationValueKey, rankClassificationNameKey, rankClassification) => {
  if (rankClassificationData && rankClassificationData.length > 0) {
    return rankClassificationData.map(v => {
      return <TreeNode obj-data={v} icon={<Radio checked={rankClassification && v[rankClassificationValueKey] === rankClassification.id}/>} value={v[rankClassificationValueKey]}
                       title={v[rankClassificationNameKey]} key={v[rankClassificationValueKey]}>
        {v.children && v.children.length > 0 && makeTreeNode(v.children, rankClassificationValueKey, rankClassificationNameKey, rankClassification)}
      </TreeNode>
    })
  }
}

const formItemLayout = {
  labelCol: {
    xs: {span: 3},
    sm: {span: 3},
  },
  wrapperCol: {
    xs: {span: 21},
    sm: {span: 21},
  },
};

const Search = ({
                  jobsText, rankText, nameText, jobsData, jobsValueKey, jobsNameKey, namePlaceholder,
                  jobsPlaceholder, rankPlaceholder, rankData, rankValueKey, rankNameKey, rankTipPlaceholder,
                  rankClassificationData, rankClassificationValueKey, rankClassificationNameKey, handleSearch,allowSelectClear
                }) => {

  const [jobs, setJobs] = useState(null);
  const [rank, setRank] = useState(null);
  const [name, setName] = useState(null);
  const [rankClassification, setRankClassification] = useState(null);


  const jobsOnSelect = (val,option) => {
    const obj = option.props['obj-data'];
    setJobs(obj);
    handleSearch(obj, name, rank, rankClassification);
  }

  const rankOnSelect = (val,option) => {
    const obj = option.props['obj-data'];
    setRank(obj);
    handleSearch(jobs, name, obj, rankClassification);
  }

  const nameSearch = (val) => {
    setName(val);
    handleSearch(jobs, val, rank, rankClassification);
  }

  const rankClassificationOnChange = (val,label,extra) => {
    if(extra.triggerNode){
      const obj = extra.triggerNode.props['obj-data'];
      setRankClassification(obj);
      handleSearch(jobs, name, rank, obj);
    }else {
      setRankClassification(null);
    }
  }

  return <Form {...formItemLayout} className={styles.search}>
    <Row>
      <Col span={12} style={{height: 50}}>
        <Form.Item label={jobsText}>
          <Select style={{width: 279}} placeholder={jobsPlaceholder} onSelect={jobsOnSelect} allowClear={allowSelectClear}>
            {jobsData && jobsData.map(v => <Option value={v[jobsValueKey]} obj-data={v}
                                                   key={v[jobsValueKey]}>{v[jobsNameKey]}</Option>)}
          </Select>
        </Form.Item>
      </Col>
      <Col span={12} style={{paddingLeft: '12px', height: 50}}>
        <Form.Item label={nameText}>
          <Input.Search placeholder={namePlaceholder} style={{width: 279}} onSearch={nameSearch}/>
        </Form.Item>
      </Col>
      <Col span={12} style={{height: 50}}>
        <Form.Item label={rankText}>
          <Select style={{width: 279}} placeholder={rankPlaceholder} onSelect={rankOnSelect} allowClear={allowSelectClear}>
            {rankData && rankData.map(v => <Option value={v[rankValueKey]} obj-data={v}
                                                   key={v[rankValueKey]}>{v[rankNameKey]}</Option>)}
          </Select>
        </Form.Item>
      </Col>
      <Col span={12} style={{paddingLeft: '12px', height: 50}}>
        <span>-</span>
        <TreeSelect
          style={{width: 279, marginLeft: 34}}
          dropdownStyle={{maxHeight: 400, overflow: 'auto'}}
          placeholder={rankTipPlaceholder}
          treeIcon={true}
          allowClear
          onChange={rankClassificationOnChange}
        >
          {makeTreeNode(rankClassificationData, rankClassificationValueKey, rankClassificationNameKey, rankClassification)}
        </TreeSelect>
      </Col>
    </Row>
  </Form>
}

export default Form.create()(Search);


src/components/contacts.js
import React, {useState, useEffect} from 'react';
import PropTypes from 'prop-types';
import {
  Col,
  Form,
  Row,
  Spin,
  Tag,
  Icon,
  message, Pagination,
} from 'antd';

import Search from './search';
import UserList from './userList';
import styles from './contacts.less';

const Contacts = (props) => {

  const {
    loading = false,
    numberColor, totalShowText, handleSearchUser, updateSelectUsers, defaultUserSelected,

    jobsText, nameText, rankText, workNumberNumber, functionText, jobsData,
    jobsValueKey, jobsNameKey, namePlaceholder, jobsPlaceholder, rankPlaceholder, rankData,
    rankValueKey, rankNameKey, rankTipPlaceholder, rankClassificationData, rankClassificationValueKey,
    rankClassificationNameKey, tableColumnsKey, tableRowKey, emptyTip,allowSelectClear,tableCheckboxDisabled
  } = props;

  /**
   * 处理查询结果为空
   */
  let {searchResult} = props;
  if (!searchResult || !searchResult.hasOwnProperty('size')) {
    searchResult = {
      records: [],
      total: 0,
      size: 9,
      current: 0,
      pages: 1,
    }
  }

  const [selectUser, setSelectUser] = useState([]);
  const [searchKey, setSearchKey] = useState(null);

  useEffect(() => {
    updateSelectUsers(defaultUserSelected);
    setSelectUser(defaultUserSelected);
  }, [defaultUserSelected, updateSelectUsers]);

  /**
   * 按照查询form中的参数查询
   * @param jobs
   * @param name
   * @param rank
   * @param rankClassification
   */
  const handleSearch = (jobs, name, rank, rankClassification) => {
    if (handleSearchUser) {
      setSearchKey({
        jobs, name, rank, rankClassification
      });
      handleSearchUser(0, jobs, name, rank, rankClassification);
    } else {
      message.error('search function not found.');
    }
  };

  /**
   * 翻页查询
   * @param page
   */
  const handlePaginationSearch = (page) => {
    if (handleSearchUser) {
      if (searchKey) {
        const {jobs, name, rank, rankClassification} = searchKey;
        handleSearchUser(page, jobs, name, rank, rankClassification);
      } else {
        handleSearchUser(page, null, null, null, null);
      }
    } else {
      message.error('search function not found.');
    }
  }

  /**
   * 生成显示的用户Tag
   * @param v
   * @return {*}
   */
  const makeUserTag = v => {
    return (
      <Tag
        key={v[tableRowKey]}
        className={styles.userTag}
        onClick={e => {
          e.preventDefault();
          unCheckUser(v);
        }}
      >
        {v[tableColumnsKey[0]]} <Icon type="close-circle" theme="filled"
                                      style={{width: 14, height: 14, color: '#E65653'}}/>
      </Tag>
    );
  };

  /**
   * 点击用户Tag时取消选择
   * @param data
   */
  const unCheckUser = data => {
    const tmp = [];
    const result = selectUser.filter(value => value[tableRowKey] !== data[tableRowKey]);
    const userList = result.concat(tmp);
    updateSelectUsers(userList);
    setSelectUser(userList);
  };

  const makeShowMsg = () => {
    const tmp = totalShowText.split('$');
    let font = '';
    let end = '';
    if (tmp.length === 2) {
      font = tmp[0];
      end = tmp[1];
    } else {
      font = totalShowText;
    }
    return (
      <div>{font} <span
        style={{color: numberColor}}>{selectUser.length}</span> {end}
      </div>
    );
  };


  return (
    // <div style={{ height: '100%',width:'690px', padding:'22px 12px 10px 12px' }}>
    <div style={{height: '100%', width: '690px'}}>
      <Spin spinning={loading}>
        <div className={styles.upside}>
          <Search jobsText={jobsText} nameText={nameText} rankText={rankText} jobsData={jobsData}
                  jobsValueKey={jobsValueKey} jobsNameKey={jobsNameKey} namePlaceholder={namePlaceholder}
                  jobsPlaceholder={jobsPlaceholder} rankPlaceholder={rankPlaceholder} rankData={rankData}
                  rankValueKey={rankValueKey} rankNameKey={rankNameKey} rankTipPlaceholder={rankTipPlaceholder}
                  rankClassificationData={rankClassificationData}
                  rankClassificationValueKey={rankClassificationValueKey}
                  rankClassificationNameKey={rankClassificationNameKey}
                  handleSearch={handleSearch} allowSelectClear={allowSelectClear}/>


          <UserList jobsText={jobsText} nameText={nameText} rankText={rankText} userData={searchResult}
                    functionText={functionText} workNumberNumber={workNumberNumber} selectUser={selectUser}
                    setSelectUser={setSelectUser} updateSelectUsers={updateSelectUsers}
                    tableColumnsKey={tableColumnsKey}
                    emptyTip={emptyTip} tableRowKey={tableRowKey} tableCheckboxDisabled={tableCheckboxDisabled}/>
        </div>
        <div className={styles.pagination}>
          <Pagination
            className={styles.pageNoe}
            simple
            current={searchResult.current || 1}
            pageSize={searchResult.size}
            total={searchResult.total}
            onChange={handlePaginationSearch}
          />
        </div>
        <Row style={{height: 150}}>
          <Col span={24} style={{paddingTop: 8, paddingBottom: 0}}>
            <Form colon={false}>
              <Form.Item className={styles.label} label={makeShowMsg()}>
                <div className={styles.resultDiv}>
                  {selectUser && selectUser.map(v => makeUserTag(v))}
                </div>
              </Form.Item>
            </Form>
          </Col>
        </Row>
      </Spin>
    </div>
  )

};

Contacts.propTypes = {
  loading: PropTypes.bool,
  searchResult: PropTypes.object,
  handleSearchUser: PropTypes.func.isRequired,
  updateSelectUsers: PropTypes.func.isRequired,
  defaultUserSelected: PropTypes.array,
  debug: PropTypes.bool,
  numberColor: PropTypes.string,
  totalShowText: PropTypes.string,

  jobsText: PropTypes.string,
  nameText: PropTypes.string,
  rankText: PropTypes.string,
  workNumberNumber: PropTypes.string,
  functionText: PropTypes.string,
  namePlaceholder: PropTypes.string,
  jobsPlaceholder: PropTypes.string,
  rankPlaceholder: PropTypes.string,
  rankTipPlaceholder: PropTypes.string,
  jobsData: PropTypes.array,
  jobsValueKey: PropTypes.string,
  jobsNameKey: PropTypes.string,
  rankData: PropTypes.array,
  rankValueKey: PropTypes.string,
  rankNameKey: PropTypes.string,
  rankClassificationData: PropTypes.array,
  rankClassificationValueKey: PropTypes.string,
  rankClassificationNameKey: PropTypes.string, tableColumnsKey: PropTypes.array,
  tableRowKey: PropTypes.string,
  emptyTip: PropTypes.string,
  allowSelectClear:PropTypes.bool,
  tableCheckboxDisabled:PropTypes.array
};

Contacts.defaultProps = {
  loading: false,
  searchResult: {
    records: [],
    total: 0,
    size: 9,
    current: 0,
    pages: 1,
  },
  defaultUserSelected: [],
  numberColor: '#E65653',
  debug: false,
  totalShowText: '共选择了$个',

  jobsText: '岗位',
  nameText: '姓名',
  rankText: '职级',
  workNumberNumber: '工号',
  functionText: '职能',
  namePlaceholder: '请输入搜索姓名',
  jobsPlaceholder: '请选择岗位',
  rankPlaceholder: '请选择职级',
  rankTipPlaceholder: '请选择职级分类或级别',
  jobsData: [],
  jobsValueKey: 'id',
  jobsNameKey: 'name',
  rankData: [],
  rankValueKey: 'id',
  rankNameKey: 'name',
  rankClassificationData: [],
  rankClassificationValueKey: 'id',
  rankClassificationNameKey: 'name',
  tableColumnsKey: ['name', 'workNumberNumber', 'fun', 'rank', 'jobs'],
  tableCheckboxDisabled:[],
  tableRowKey: 'id',
  emptyTip: '什么都没有哦~',
  allowSelectClear:true
};

export default Contacts;


src/mockData.js
/**
 * Created : vincent
 * Date : 2019-07-10  16:57
 * Email : wangxiao@wafersystems.com
 */

export const deptTree = [{
  id: 1,
  parentId: 0,
  children: [
    {
      id: 3,
      parentId: 1,
      children: [
        {
          id: 4,
          parentId: 3,
          children: [
            {
              id: 5,
              parentId: 4,
              children: [],
              name: '院校农信',
            },
            {
              id: 6,
              parentId: 4,
              children: [],
              name: '院校农信123',
            },
          ],
          name: '高新农信',
        },
        {
          id: 14,
          parentId: 3,
          children: [
            {
              id: 15,
              parentId: 14,
              children: [],
              name: '院校农信B',
            },
            {
              id: 16,
              parentId: 14,
              children: [],
              name: '院校农信A',
            },
          ],
          name: '高新农信C',
        },
      ],
      name: '潍坊农信',
    },
  ],
  name: '山东农信',
}];

const tmp = []

for (let i = 0; i < 9; i++) {
  tmp.push({
    userId: 1,
    id: i,
    name: 'admin',
    password: '$2a$10$QOfWxxFyAMmEEmnuw9UI/..1s4B4eF/u9PzE2ZaGO.ij9YfmcUy.u',
    salt: null,
    wxOpenid: 'o_0FT0uyg_H1vVy2H0JpSwlVGhWQ',
    qqOpenid: null,
    createTime: '2018-04-20 07:15:18',
    updateTime: '2019-03-12 16:04:42',
    jobs: 'a@a.com',
    delFlag: '0',
    lockFlag: '0',
    rank: '17034642888',
    avatar: 'lengleng-0d2a7b025da14d8d93f595b3fa082d82.jpg',
    workNumberNumber: 1,
    tenantId: 1,
    fun: '真理部',
    roleList: [
      {
        roleId: 1,
        roleName: '管理员',
        roleCode: 'ROLE_ADMIN',
        roleDesc: '管理员',
        dsType: 2,
        dsScope: '2',
        createTime: '2017-10-29 15:45:51',
        updateTime: '2018-12-26 14:09:11',
        delFlag: '0',
      },
    ],
  },)
}

export const userList = {
  records: tmp,
  total: 11,
  size: 10,
  current: 1,
  searchCount: true,
  pages: 1,
};

export const us = {
  "records": [{
    "adAccount": "cam_gan",
    "accountName": "3160",
    "chsName": "甘露",
    "engName": "Cam",
    "gradeName": "高级主任",
    "gradeType": "主任级",
    "positionType": 0,
    "positionName": "副总裁助理（高级主任）",
    "orgName": "市场"
  }, {
    "adAccount": "yuki_kong",
    "accountName": "1104",
    "chsName": "江雪琦",
    "engName": "Xueqi",
    "gradeName": "高级主任",
    "gradeType": "主任级",
    "positionType": 0,
    "positionName": "无限极全球行政总裁助理（高级主任）",
    "orgName": "无限极全球"
  }, {
    "adAccount": "genies_ou",
    "accountName": "1250",
    "chsName": "欧婕寰",
    "engName": "Genies",
    "gradeName": "高级主任",
    "gradeType": "主任级",
    "positionType": 0,
    "positionName": "副总裁助理（高级主任）",
    "orgName": "产品供应"
  }, {
    "adAccount": "qiong_zeng",
    "accountName": "5085",
    "chsName": "曾琼",
    "engName": "Judy",
    "gradeName": "主任",
    "gradeType": "主任级",
    "positionType": 0,
    "positionName": "高级副总裁助理（主任）",
    "orgName": "无限极全球"
  }, {
    "adAccount": "XiaoQing.Dong",
    "accountName": "5807",
    "chsName": "董晓青",
    "engName": "Carol",
    "gradeName": "副主任",
    "gradeType": "主任级",
    "positionType": 0,
    "positionName": "行政支持副主任",
    "orgName": "广州无限极广场"
  }, {
    "adAccount": "amy_liu",
    "accountName": "5601",
    "chsName": "刘淡真",
    "engName": "Amy",
    "gradeName": "高级主任",
    "gradeType": "主任级",
    "positionType": 0,
    "positionName": "项目支持高级主任",
    "orgName": "快消品售后服务"
  }, {
    "adAccount": "huiyi_liu",
    "accountName": "5710",
    "chsName": "刘慧仪",
    "engName": "Chris",
    "gradeName": "高级主任",
    "gradeType": "主任级",
    "positionType": 0,
    "positionName": "高级副总裁助理（高级主任）",
    "orgName": "区域管理及外事"
  }, {
    "adAccount": "Yoyo.Zhang",
    "accountName": "5855",
    "chsName": "张碧怡",
    "engName": "Yoyo",
    "gradeName": "高级主任",
    "gradeType": "主任级",
    "positionType": 0,
    "positionName": "副总裁助理（高级主任）",
    "orgName": "行政支援"
  }, {
    "adAccount": "Vanessa.Siu",
    "accountName": "0008495",
    "chsName": "蕭子君",
    "engName": "Vanessa",
    "gradeName": "主任",
    "gradeType": "",
    "positionType": 0,
    "positionName": "总裁助理",
    "orgName": "爽乐行政总裁办公室"
  }], "total": 15, "size": 9, "current": 1, "searchCount": true, "pages": 2
}

export const defaultSelect = [{
  "userId": 350,
  "username": "Virginia.Zhou",
  "password": null,
  "createTime": 1603856178000,
  "updateTime": 1604459977000,
  "delFlag": "0",
  "lockFlag": "0",
  "phone": "13731115114",
  "avatar": "",
  "deptId": 1668,
  "tenantId": 1,
  "realName": "周紫竹",
  "realNameEn": "Virginia",
  "firstName": "",
  "lastName": "",
  "namePinyin": "zhouzizhu",
  "gender": 0,
  "position": "高级职员",
  "positionEn": "",
  "extraJobFlag": 0,
  "mail": "virginia.zhou@infinitus-int.com",
  "homePhone": "0311-67307755",
  "ord": 11236,
  "entrySource": "infinitus",
  "entryid": "",
  "entrypid": "9574",
  "workNumber": "1250",
  "cardNumber": "",
  "lang": "",
  "defaultZone": null,
  "phoneStatus": null,
  "mailStatus": null,
  "timeZone": "Asia/Shanghai"
}, {
  "userId": 351,
  "username": "Super.Shu",
  "password": null,
  "createTime": 1603856178000,
  "updateTime": 1604459982000,
  "delFlag": "0",
  "lockFlag": "0",
  "phone": "13229406055",
  "avatar": "",
  "deptId": 1797,
  "tenantId": 1,
  "realName": "舒鹏",
  "realNameEn": "Super",
  "firstName": "",
  "lastName": "",
  "namePinyin": "shupeng",
  "gender": 1,
  "position": "-",
  "positionEn": "",
  "extraJobFlag": 0,
  "mail": "super.shu@infinitus-int.com",
  "homePhone": "02080734702",
  "ord": 13788,
  "entrySource": "infinitus",
  "entryid": "",
  "entrypid": "26279856",
  "workNumber": "5085",
  "cardNumber": "",
  "lang": "",
  "defaultZone": null,
  "phoneStatus": null,
  "mailStatus": null,
  "timeZone": "Asia/Shanghai"
}]

export const tree = [{
  "name": "主席兼行政总裁",
  "type": 0,
  "id": -952998653,
  "parentId": -1,
}, {
  "name": "副总裁级",
  "type": 0,
  "id": 653471410,
  "parentId": -1,
  "children": [{"name": "高级副总裁", "type": 1, "id": -1059340954, "parentId": 653471410, "children": []}, {
    "name": "副总裁",
    "type": 1,
    "id": 21078677,
    "parentId": 653471410,
    "children": []
  }, {"name": "集团副总裁", "type": 1, "id": 1970015289, "parentId": 653471410, "children": []}, {
    "name": "集团高级副总裁",
    "type": 1,
    "id": -736997878,
    "parentId": 653471410,
    "children": []
  }, {"name": "行政总裁", "type": 1, "id": 1065174425, "parentId": 653471410, "children": []}]
}, {
  "name": "非编制人员",
  "type": 0,
  "id": -1878174948,
  "parentId": -1,
}, {
  "name": "职员级",
  "type": 0,
  "id": 32264859,
  "parentId": -1,
  "children": [{"name": "高级职员", "type": 1, "id": 1213113499, "parentId": 32264859, "children": []}, {
    "name": "T2",
    "type": 1,
    "id": 2654,
    "parentId": 32264859,
    "children": []
  }, {"name": "职员", "type": 1, "id": 1039756, "parentId": 32264859, "children": []}, {
    "name": "T1",
    "type": 1,
    "id": 2653,
    "parentId": 32264859,
    "children": []
  }, {"name": "P2", "type": 1, "id": 2530, "parentId": 32264859, "children": []}]
}, {
  "name": "主任级",
  "type": 0,
  "id": 19905159,
  "parentId": -1,
  "children": [{"name": "P4", "type": 1, "id": 2532, "parentId": 19905159, "children": []}, {
    "name": "主任",
    "type": 1,
    "id": 641056,
    "parentId": 19905159,
    "children": []
  }, {"name": "P3", "type": 1, "id": 2531, "parentId": 19905159, "children": []}, {
    "name": "副主任",
    "type": 1,
    "id": 20921039,
    "parentId": 19905159,
    "children": []
  }, {"name": "主任审批", "type": 1, "id": 616807192, "parentId": 19905159, "children": []}, {
    "name": "T4",
    "type": 1,
    "id": 2656,
    "parentId": 19905159,
    "children": []
  }, {"name": "T5", "type": 1, "id": 2657, "parentId": 19905159, "children": []}, {
    "name": "高级主任",
    "type": 1,
    "id": 1212714799,
    "parentId": 19905159,
    "children": []
  }, {"name": "T3", "type": 1, "id": 2655, "parentId": 19905159, "children": []}, {
    "name": "P5",
    "type": 1,
    "id": 2533,
    "parentId": 19905159,
    "children": []
  }]
}, {
  "name": "工人级",
  "type": 0,
  "id": 23756754,
  "parentId": -1,
  "children": [{"name": "工人", "type": 1, "id": 765301, "parentId": 23756754, "children": []}]
}, {
  "name": "总监级",
  "type": 0,
  "id": 24649585,
  "parentId": -1,
  "children": [{"name": "代理总监", "type": 1, "id": 630966969, "parentId": 24649585, "children": []}, {
    "name": "总监",
    "type": 1,
    "id": 794102,
    "parentId": 24649585,
    "children": []
  }, {"name": "策略顾问", "type": 1, "id": 970750527, "parentId": 24649585, "children": []}, {
    "name": "高级顾问",
    "type": 1,
    "id": 1213322303,
    "parentId": 24649585,
    "children": []
  }]
}, {
  "name": "经理级",
  "type": 0,
  "id": 32150128,
  "parentId": -1,
  "children": [{"name": "P7", "type": 1, "id": 2535, "parentId": 32150128, "children": []}, {
    "name": "经理",
    "type": 1,
    "id": 1036055,
    "parentId": 32150128,
    "children": []
  }, {"name": "T6", "type": 1, "id": 2658, "parentId": 32150128, "children": []}, {
    "name": "顾问",
    "type": 1,
    "id": 1248560,
    "parentId": 32150128,
    "children": []
  }, {"name": "副经理", "type": 1, "id": 21316038, "parentId": 32150128, "children": []}, {
    "name": "P6",
    "type": 1,
    "id": 2534,
    "parentId": 32150128,
    "children": []
  }, {"name": "高级经理", "type": 1, "id": 1213109798, "parentId": 32150128, "children": []}, {
    "name": "P8",
    "type": 1,
    "id": 2536,
    "parentId": 32150128,
    "children": []
  }, {"name": "T7", "type": 1, "id": 2659, "parentId": 32150128, "children": []}, {
    "name": "负责人L",
    "type": 1,
    "id": 1111606198,
    "parentId": 32150128,
    "children": []
  }, {"name": "T8", "type": 1, "id": 2660, "parentId": 32150128, "children": []}]
}, {
  "name": "总经理级",
  "type": 0,
  "id": 766051413,
  "parentId": -1,
  "children": [{"name": "总经理", "type": 1, "id": 24710290, "parentId": 766051413, "children": []}, {
    "name": "首席工程师",
    "type": 1,
    "id": -1601383605,
    "parentId": 766051413,
    "children": []
  }, {"name": "资深顾问", "type": 1, "id": 1105657629, "parentId": 766051413, "children": []}, {
    "name": "T10",
    "type": 1,
    "id": 82291,
    "parentId": 766051413,
    "children": []
  }]
}]
